src/PerfView/PerfViewData.cs

```diff
@@ -3388,6 +3388,447 @@ private void CreateCSVFile(List<AspNetRequest> requests, string csvFileName)
         #endregion
     }
 
+    /// <summary>
+    /// Generates an ASP.NET Core Stats HTML page
+    /// </summary>
+    public class PerfViewAspNetCoreStats : PerfViewHtmlReport
+    {
+        private const string _hostingProvider = "Microsoft.AspNetCore.Hosting";
+        private const string _eventManifestData = "ManifestData";
+        private const string _eventRequestStart = "Request/Start";
+        private const string _eventRequestStop = "Request/Stop";
+        private const string _eventHostStart = "Host/Start";
+        private const string _eventHostStop = "Host/Stop";
+        private const string _eventUnhandledException = "UnhandledException";
+        private const string _eventServerReady = "ServerReady";
+        //private const string _iloggerProvider = "Microsoft-Extensions-Logging";
+        //private const string _iloggerProviderFormattedMessage = "Microsoft-Extensions-Logging/FormattedMessage";
+
+        // master requests dictionary
+        private Dictionary<Guid, ANCHostingRequest> requests = new Dictionary<Guid, ANCHostingRequest>();
+
+        // holds non-request events like host/start and stop, etc.
+        private List<OtherHostingEvent> otherEvents = new List<OtherHostingEvent>();
+
+        public PerfViewAspNetCoreStats(PerfViewFile dataFile) : base(dataFile, "ASP.NET Core Stats") { }
+
+        protected override void WriteHtmlBody(TraceLog dataFile, TextWriter outputWriter, string fileName, TextWriter log)
+        {
+            outputWriter.WriteLine("<H2>ASP.NET Core Request Statistics</H2>");
+
+            // request/stop does not contain status code or anything like that
+            // maybe add ILogger events, if available, to enhance these logs?
+            // also down the road if both ILogger and MS-ANC-Hosting events are available, use both and integrate them?
+            /* 
+             * some possible events (ref: https://source.dot.net/#Microsoft.AspNetCore.Hosting/Internal/HostingEventSource.cs):
+             * Request/Start and Request/Stop
+             * ManifestData
+             * Host/Start and Host/Stop
+             * UnhandledException
+             * ServerReady
+             */
+
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+
+            // using the pull/foreach model for iterating through the TraceLog events
+            foreach (TraceEvent traceEvent in dataFile.Events)
+            {
+                if (traceEvent.ProviderName.Equals(_hostingProvider))
```

Comment:
Recommend that rather than using `foreach` here, you call `dataFile.Events.GetSource()` and then call `AddCallbackForProviderEvent`, passing in the provider name, and `null` for the event name.  This will give you all events for the provider, but will do in a more efficient manner than iterating over all events in the trace.

Reply:
Yes, I would still recommend using the callback model.  This allows the dispatch machinery to bail out earlier and not do as much work for events that you don't care about, which are likely to be a large chunk of the total events.

Reply:
understood and will do

Reply:
done in latest commit